rfcs/README.md

```diff
@@ -0,0 +1,98 @@
+# oneDPL Design Documents/RFCs
+
+The Request for Comments (RFC) process intends to:
+
+- Propose and discuss ideas for the library evolution
+- Communicate anticipated library-wide changes
+- Collect feedback before implementation
+- Increase transparency in decision-making 
+- Align different teams involved in the library development
+
+Most modifications or new features will naturally start as a part of a 
+GitHub issue or discussion. Small changes do not require a formal RFC. 
+However, if the issue or discussion results in an idea for a significant 
+change or new feature that affects the library's public API or architecture, 
+we recommend creating a formal document that provides
+a detailed description and design of the proposed feature.
+
+This directory contains design documents (RFCs) approved, 
+implemented, or rejected for implementation.
+
+## RFC Directory Structure
+
+The design documents are stored in the `rfcs` directory, each placed 
+in a subdirectory under an `rfcs/<state>` directory. The possible states are:
+
+1. Proposed
+2. Experimental
+3. Supported
```

Comment:
Does it make sense to have either a subdirectory within "Supported" named "Extension", or another top level directory for "Extension"?  

I'm wondering about features which might be extensions either because they might rely upon hardware / compiler specific functionality or perhaps features which we want to show have more stability / robustness but are are ready for the specification for some reason. We've discussed labelling extensions in the past and it may make sense to codify that with some structure.

The time limit on experimental got me thinking about this.  Both ranges and async APIs could've fit into such a category (until recently in the case of Ranges API).  They have been reasonably stable and even depended on internally from non-experimental code, but were still experimental.

Reply:
I prefer to wait for a precedent or maybe two, and decide based on that experience.

Reply:
Sure, I understand why we may want this only in rare cases and not advertise it. This will avoid the temptation of using it as an escape valve for experimental projects which stall rather than pushing for graduation to the specification.

I'm OK with leaving it out until we see it is necessary.